### PR TITLE
Validation model specs

### PR DESCRIPTION
**Pull request by wacoo**
- Add validations to models
- Add rspec tests to  validated attributes
- Add rspec tests to model methods

### CODE REVIEW

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  end
+
+  it 'returns the three most recent users' do
+    user1 = User.create(name: 'John', photo: 'https://example.com/photo1.jpg', bio: 'Bio 1', posts_counter: 3)
+    user2 = User.create(name: 'Jane', photo: 'https://example.com/photo2.jpg', bio: 'Bio 2', posts_counter: 1)
+    user3 = User.create(name: 'Alice', photo: 'https://example.com/photo3.jpg', bio: 'Bio 3', posts_counter: 2)
+
+    recent_users = User.recent_three.to_a
+    expect(recent_users).to match_array([user1, user3, user2])
+  end
```
Comment (V-Blaze) [inline spec/models/user_spec.rb]: - You have done great so far :+1: But there are a few requirements you need to meet before moving to the next milestone. :+1: 

- [ ] All your tests are passing correctly, and your tests seem to cover all the validations and methods, but this last test in the user model is not correct. 
The test is meant to be on a `three_most_recent_post` method in the user's model. Looking at the model, it seems the method you implemented there initially is a method that returns `3 recent users` but that's not the method that was in the requirement, the correct method is `A method that returns the 3 most recent posts for a given user.`. (this should have been spotted and mentioned by the previous reviewer of that milestone.).

I'll suggest you go to your user model and implement the correct method, then return to your test file and rewrite the test for this new correct method. :+1: :100: :+1: Here's a screenshot from the requirement of that milestone. :+1: 
- - Screenshot reference:
Selection_024

File: spec/models/user_spec.rb [hunk 2]
```diff
@@ -0,0 +1,44 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  # let!(:user1) { User.create(name: 'John', posts_counter: 3) }
+  # let!(:user2) { User.create(name: 'Jane', posts_counter: 1) }
+  # let!(:user3) { User.create(name: 'Alice', posts_counter: 2) }
```
Comment (V-Blaze) [inline spec/models/user_spec.rb]: - [ ] Follow our list of [best practices for Ruby](https://github.com/microverseinc/curriculum-ruby/blob/main/articles/ruby_best_practices.md), You are not meant to leave un-used code as a comment in your codebase. I'll suggest you remove these as leaving unused code as a comment in your codebase makes your code look a bit unprofessional :+1: :smile: :wave:

**Review by V-Blaze — CHANGES_REQUESTED** [review]
## Hi @wacoo 

## Required changes 🖊️ 
# Almost there 🟡 🟦 🟨 

giphy


Great work so far, You have come a long way, keep moving forward👏 👍  
You've done well implementing the project requirements, but there are still some issues I think you need to improve on. 😄 👋 


 ## Highlights
- `User and Post` model validation implemented correctly ✔
-  All tests are passing👋 
-  Correct git flow✔

 ## Required Changes ♻️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mukhammadrizooff — APPROVED** [review]
Hi @wacoo, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉

 

## Highlights
- No linter errors ✔️ 
- Correct GitFlow ✔️ 
- Good implementations ✔️
- Descriptive `README` file ✔️

image

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**